src/main/java/io/cryostat/net/web/http/api/v1/RecordingsGetHandler.java

```diff
@@ -111,48 +88,12 @@ public boolean isAsync() {
 
     @Override
     public void handleAuthenticated(RoutingContext ctx) throws Exception {
-        if (!fs.exists(savedRecordingsPath)) {
-            throw new HttpStatusException(
-                    501,
-                    String.format(
-                            "Archive path %s does not exist", savedRecordingsPath.toString()));
+        try {
+            List<ArchivedRecordingInfo> result = recordingArchiveHelper.getRecordings().get();
+            ctx.response().setStatusCode(200);
+            ctx.response().end(gson.toJson(result));
+        } catch (ArchivePathException e) {
+            throw new HttpStatusException(501, e.getMessage());
```

Comment:
Should `e` be passed to the exception constructor as well to establish the causal link between them?